// chat-view/esm2015/lib/chat-view.component.js
import * as tslib_1 from "tslib";
import { query, stagger, style, transition, trigger, useAnimation } from '@angular/animations';
import { Component, ElementRef, EventEmitter, Input, Output, ViewChild } from '@angular/core';
import { messagePushAnimation } from './animations/animations';
let ChatViewComponent = 
/**
 * This is only componet that needs to be added while integrating chat-view library.
 * Parent component of chat messages, chat header and chat input
 * Input : Array of ReceivedMessage and SentMessage
 * Input : IChatViewOptions : Set style properties of ReceivedMessage, SentMessage, header, chat input, quick replies
 * Output: onOptionClicked event emitter : Triggered when user clicks one of the quick replies
 * Output: onInput event emitter : Triggered when user clicks of send button
 *
 * @export
 * @class ChatViewComponent
 * @implements {OnInit}
 */
class ChatViewComponent {
    constructor() {
        /**
         * Input messages (recieved or sent) array passed to component.
         * @type {(ReceivedMessage[] | SentMessage[])}
         * @memberof ChatViewComponent
         */
        this.messages = [];
        /**
         * Input property to set styling options to chat-view compoenents
         * @type {IChatViewOptions}
         * @memberof ChatViewComponent
         */
        this.options = {
            //Set background color and text color of SentMessage
            sentMessage: {
                textColor: 'white',
                backgroundColor: '#c7ac7b',
            },
            //Set background color and text color of ReceivedMessage
            receivedMessage: {
                textColor: 'black',
                backgroundColor: 'white',
            },
            //Set background color and text color of ChatHeader
            header: {
                textColor: 'white',
                backgroundColor: '#71a7de'
            },
            //Set border color, text color, max input length, send icon color and background color of input box
            inputBox: {
                borderColor: 'gray',
                textColor: 'black',
                maxLength: 50,
                sendIconColor: '#5e8dbd',
                backgroundColor: '#e1faf8'
            },
            //Set text color, background color and border color of quick replies
            quickReplies: {
                textColor: 'white',
                backgroundColor: '#c7ac7b',
                borderColor: '#c7ac7b'
            },
            //Set background color of chatlist pane
            chatlistBackgroundColor: '#e1faf8',
        };
        /**
         * Event emitter for quick replies are clicked
         * @memberof ChatViewComponent
         */
        this.onOptionClicked = new EventEmitter();
        /**
         * Event emitter when send button is clicked
         * @memberof ChatViewComponent
         */
        this.onInput = new EventEmitter();
        //messages array to keep track of previous messages.
        //This is used in ngDoCheck() top scroll chatview if new message is added.
        this.oldMessages = [];
    }
    /**
     * Event emitter when send button is clicked.
     * Parent copmponent will recived users query. Then it can be added in sent message array.
     * @param {*} text
     * @memberof ChatViewComponent
     */
    sendMessage(text) {
        this.onInput.emit(text);
    }
    /**
     * Event emitter when quick replies are clicked.
     * Parent copmponent will recived selected option. Then it can be added in sent message array.
     * @param {*} text
     * @memberof ChatViewComponent
     */
    optionClicked(option) {
        this.onInput.emit(option);
    }
    /**
     * To scroll chatlist to the bottom
     * @memberof ChatViewComponent
     */
    scrollToBottom() {
        setTimeout(() => {
            this.chatlist.nativeElement.scrollTop = this.chatlist.nativeElement.scrollHeight;
        }, 0);
    }
    /**
     * After every change detection cycle, check if new message is added in message array.
     * If yes, then scroll chat list to bottom.
     * @memberof ChatViewComponent
     */
    ngDoCheck() {
        if (this.messages && this.messages.length != this.oldMessages.length) {
            this.scrollToBottom();
        }
        this.oldMessages = [...this.messages];
    }
};
tslib_1.__decorate([
    ViewChild('chatlist', { static: false }),
    tslib_1.__metadata("design:type", ElementRef
    /**
     * Input messages (recieved or sent) array passed to component.
     * @type {(ReceivedMessage[] | SentMessage[])}
     * @memberof ChatViewComponent
     */
    )
], ChatViewComponent.prototype, "chatlist", void 0);
tslib_1.__decorate([
    Input('messages'),
    tslib_1.__metadata("design:type", Array)
], ChatViewComponent.prototype, "messages", void 0);
tslib_1.__decorate([
    Input('options'),
    tslib_1.__metadata("design:type", Object)
], ChatViewComponent.prototype, "options", void 0);
tslib_1.__decorate([
    Output('onOptionClicked'),
    tslib_1.__metadata("design:type", Object)
], ChatViewComponent.prototype, "onOptionClicked", void 0);
tslib_1.__decorate([
    Output('onInput'),
    tslib_1.__metadata("design:type", Object)
], ChatViewComponent.prototype, "onInput", void 0);
ChatViewComponent = tslib_1.__decorate([
    Component({
        selector: 'chat-view',
        template: "<div class=\"chat-container\">\n    <chat-header title=\"My Chat View\" [headerOptions]=\"options.header\"></chat-header>\n    <div #chatlist class=\"chat-list-wrapper\"\n      [@messageAddedAnimation]=\"messages.length\"\n      [style.backgroundColor]=\"options.chatlistBackgroundColor\"\n      >\n        <div  class=\"chat-list-item\" *ngFor=\"let message of messages;let i=index\">\n\n            <!--received Message-->\n            <div *ngIf=\"message.getCategory()=='received'\">\n                <received-message [message]=\"message\"\n                (onOptionClicked)=\"optionClicked($event)\" \n                [receivedMessageOptions]=\"options.receivedMessage\"\n                [quickRepliesOptions]=\"options.quickReplies\"\n                ></received-message>\n            </div>\n\n            <!--Sent Message-->\n            <div *ngIf=\"message.getCategory()=='sent'\">\n                <sent-message [message]=\"message\"\n                [sentMessageOptions]=\"options.sentMessage\"\n                ></sent-message>\n            </div>\n        </div>\n    </div>\n    <user-input (onInput)=\"sendMessage($event)\"\n    [options]=\"options.inputBox\"></user-input>\n</div>",
        animations: [
            trigger('messageAddedAnimation', [
                transition(':enter, * => 0, * => -1', []),
                transition(':increment', [
                    query(':enter', [
                        style({ opacity: 0 }),
                        stagger(100, [
                            useAnimation(messagePushAnimation, {
                                params: {
                                    opacity: 1,
                                    translateY: 10,
                                    time: '0.2s'
                                }
                            })
                        ]),
                    ], { optional: true })
                ]),
            ]),
        ],
        styles: ["div{color:red}.chat-container{width:100%;height:100%;display:-webkit-box;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:end;justify-content:flex-end}.chat-list-wrapper{-webkit-box-flex:1;flex:1 1 auto;overflow-y:scroll}.chat-list-wrapper .chat-list-item:first-child{margin-top:5px}.chat-list-wrapper .chat-list-item,.chat-list-wrapper received-message sent-message{margin-bottom:5px}chat-header{box-shadow:0 5px 11px #dbd9d9be;z-index:1}user-input{-webkit-box-flex:0;flex:0 0 auto;background:0 0;z-index:1}received-message,sent-message{display:block;width:100%}"]
    })
    /**
     * This is only componet that needs to be added while integrating chat-view library.
     * Parent component of chat messages, chat header and chat input
     * Input : Array of ReceivedMessage and SentMessage
     * Input : IChatViewOptions : Set style properties of ReceivedMessage, SentMessage, header, chat input, quick replies
     * Output: onOptionClicked event emitter : Triggered when user clicks one of the quick replies
     * Output: onInput event emitter : Triggered when user clicks of send button
     *
     * @export
     * @class ChatViewComponent
     * @implements {OnInit}
     */
    ,
    tslib_1.__metadata("design:paramtypes", [])
], ChatViewComponent);
export { ChatViewComponent };
