// chat-view/esm5/lib/components/header/header.component.js
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
var ChatHeaderComponent = /** @class */ (function () {
    /**
     *Creates an instance of ChatHeaderComponent.
     * @memberof ChatHeaderComponent
     */
    function ChatHeaderComponent() {
        this.title = 'Chat View';
    }
    ChatHeaderComponent.prototype.ngOnInit = function () {
    };
    tslib_1.__decorate([
        Input('title'),
        tslib_1.__metadata("design:type", Object)
    ], ChatHeaderComponent.prototype, "title", void 0);
    tslib_1.__decorate([
        Input('headerOptions'),
        tslib_1.__metadata("design:type", Object)
    ], ChatHeaderComponent.prototype, "headerOptions", void 0);
    ChatHeaderComponent = tslib_1.__decorate([
        Component({
            selector: 'chat-header',
            template: "<div class=\"chat-header-wrapper\" [style.backgroundColor]=\"headerOptions?.backgroundColor\">\n    <div class=\"chat-header-title\" [style.color]=\"headerOptions?.textColor\">\n        {{title}}\n    </div>\n</div>",
            styles: [".chat-header-wrapper{background-color:#2196f3;height:40px;display:-webkit-box;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;justify-content:space-around}.chat-header-title{text-align:center;vertical-align:middle;color:#fff}"]
        }),
        tslib_1.__metadata("design:paramtypes", [])
    ], ChatHeaderComponent);
    return ChatHeaderComponent;
}());
export { ChatHeaderComponent };
