// chat-view/bundles/chat-view.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/animations'), require('@angular/common'), require('@angular/forms'), require('@angular/platform-browser')) :
    typeof define === 'function' && define.amd ? define('chat-view', ['exports', '@angular/core', '@angular/animations', '@angular/common', '@angular/forms', '@angular/platform-browser'], factory) :
    (global = global || self, factory(global['chat-view'] = {}, global.ng.core, global.ng.animations, global.ng.common, global.ng.forms, global.ng.platformBrowser));
}(this, (function (exports, core, animations, common, forms, platformBrowser) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m) return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    var ChatViewService = /** @class */ (function () {
        function ChatViewService() {
        }
        ChatViewService.ngInjectableDef = core.ɵɵdefineInjectable({ factory: function ChatViewService_Factory() { return new ChatViewService(); }, token: ChatViewService, providedIn: "root" });
        ChatViewService = __decorate([
            core.Injectable({
                providedIn: 'root'
            }),
            __metadata("design:paramtypes", [])
        ], ChatViewService);
        return ChatViewService;
    }());

    var messagePushAnimation = animations.animation([
        animations.style({
            opacity: '{{ opacity }}',
            transform: 'translateY({{translateY}}%)'
        }),
        animations.animate('{{ time }}')
    ]);

    var ChatViewComponent = /** @class */ (function () {
        function ChatViewComponent() {
            /**
             * Input messages (recieved or sent) array passed to component.
             * @type {(ReceivedMessage[] | SentMessage[])}
             * @memberof ChatViewComponent
             */
            this.messages = [];
            /**
             * Input property to set styling options to chat-view compoenents
             * @type {IChatViewOptions}
             * @memberof ChatViewComponent
             */
            this.options = {
                //Set background color and text color of SentMessage
                sentMessage: {
                    textColor: 'white',
                    backgroundColor: '#c7ac7b',
                },
                //Set background color and text color of ReceivedMessage
                receivedMessage: {
                    textColor: 'black',
                    backgroundColor: 'white',
                },
                //Set background color and text color of ChatHeader
                header: {
                    textColor: 'white',
                    backgroundColor: '#71a7de'
                },
                //Set border color, text color, max input length, send icon color and background color of input box
                inputBox: {
                    borderColor: 'gray',
                    textColor: 'black',
                    maxLength: 50,
                    sendIconColor: '#5e8dbd',
                    backgroundColor: '#e1faf8'
                },
                //Set text color, background color and border color of quick replies
                quickReplies: {
                    textColor: 'white',
                    backgroundColor: '#c7ac7b',
                    borderColor: '#c7ac7b'
                },
                //Set background color of chatlist pane
                chatlistBackgroundColor: '#e1faf8',
            };
            /**
             * Event emitter for quick replies are clicked
             * @memberof ChatViewComponent
             */
            this.onOptionClicked = new core.EventEmitter();
            /**
             * Event emitter when send button is clicked
             * @memberof ChatViewComponent
             */
            this.onInput = new core.EventEmitter();
            //messages array to keep track of previous messages.
            //This is used in ngDoCheck() top scroll chatview if new message is added.
            this.oldMessages = [];
        }
        /**
         * Event emitter when send button is clicked.
         * Parent copmponent will recived users query. Then it can be added in sent message array.
         * @param {*} text
         * @memberof ChatViewComponent
         */
        ChatViewComponent.prototype.sendMessage = function (text) {
            this.onInput.emit(text);
        };
        /**
         * Event emitter when quick replies are clicked.
         * Parent copmponent will recived selected option. Then it can be added in sent message array.
         * @param {*} text
         * @memberof ChatViewComponent
         */
        ChatViewComponent.prototype.optionClicked = function (option) {
            this.onInput.emit(option);
        };
        /**
         * To scroll chatlist to the bottom
         * @memberof ChatViewComponent
         */
        ChatViewComponent.prototype.scrollToBottom = function () {
            var _this = this;
            setTimeout(function () {
                _this.chatlist.nativeElement.scrollTop = _this.chatlist.nativeElement.scrollHeight;
            }, 0);
        };
        /**
         * After every change detection cycle, check if new message is added in message array.
         * If yes, then scroll chat list to bottom.
         * @memberof ChatViewComponent
         */
        ChatViewComponent.prototype.ngDoCheck = function () {
            if (this.messages && this.messages.length != this.oldMessages.length) {
                this.scrollToBottom();
            }
            this.oldMessages = __spread(this.messages);
        };
        __decorate([
            core.ViewChild('chatlist', { static: false }),
            __metadata("design:type", core.ElementRef
            /**
             * Input messages (recieved or sent) array passed to component.
             * @type {(ReceivedMessage[] | SentMessage[])}
             * @memberof ChatViewComponent
             */
            )
        ], ChatViewComponent.prototype, "chatlist", void 0);
        __decorate([
            core.Input('messages'),
            __metadata("design:type", Array)
        ], ChatViewComponent.prototype, "messages", void 0);
        __decorate([
            core.Input('options'),
            __metadata("design:type", Object)
        ], ChatViewComponent.prototype, "options", void 0);
        __decorate([
            core.Output('onOptionClicked'),
            __metadata("design:type", Object)
        ], ChatViewComponent.prototype, "onOptionClicked", void 0);
        __decorate([
            core.Output('onInput'),
            __metadata("design:type", Object)
        ], ChatViewComponent.prototype, "onInput", void 0);
        ChatViewComponent = __decorate([
            core.Component({
                selector: 'chat-view',
                template: "<div class=\"chat-container\">\n    <chat-header title=\"My Chat View\" [headerOptions]=\"options.header\"></chat-header>\n    <div #chatlist class=\"chat-list-wrapper\"\n      [@messageAddedAnimation]=\"messages.length\"\n      [style.backgroundColor]=\"options.chatlistBackgroundColor\"\n      >\n        <div  class=\"chat-list-item\" *ngFor=\"let message of messages;let i=index\">\n\n            <!--received Message-->\n            <div *ngIf=\"message.getCategory()=='received'\">\n                <received-message [message]=\"message\"\n                (onOptionClicked)=\"optionClicked($event)\" \n                [receivedMessageOptions]=\"options.receivedMessage\"\n                [quickRepliesOptions]=\"options.quickReplies\"\n                ></received-message>\n            </div>\n\n            <!--Sent Message-->\n            <div *ngIf=\"message.getCategory()=='sent'\">\n                <sent-message [message]=\"message\"\n                [sentMessageOptions]=\"options.sentMessage\"\n                ></sent-message>\n            </div>\n        </div>\n    </div>\n    <user-input (onInput)=\"sendMessage($event)\"\n    [options]=\"options.inputBox\"></user-input>\n</div>",
                animations: [
                    animations.trigger('messageAddedAnimation', [
                        animations.transition(':enter, * => 0, * => -1', []),
                        animations.transition(':increment', [
                            animations.query(':enter', [
                                animations.style({ opacity: 0 }),
                                animations.stagger(100, [
                                    animations.useAnimation(messagePushAnimation, {
                                        params: {
                                            opacity: 1,
                                            translateY: 10,
                                            time: '0.2s'
                                        }
                                    })
                                ]),
                            ], { optional: true })
                        ]),
                    ]),
                ],
                styles: ["div{color:red}.chat-container{width:100%;height:100%;display:-webkit-box;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:end;justify-content:flex-end}.chat-list-wrapper{-webkit-box-flex:1;flex:1 1 auto;overflow-y:scroll}.chat-list-wrapper .chat-list-item:first-child{margin-top:5px}.chat-list-wrapper .chat-list-item,.chat-list-wrapper received-message sent-message{margin-bottom:5px}chat-header{box-shadow:0 5px 11px #dbd9d9be;z-index:1}user-input{-webkit-box-flex:0;flex:0 0 auto;background:0 0;z-index:1}received-message,sent-message{display:block;width:100%}"]
            })
            /**
             * This is only componet that needs to be added while integrating chat-view library.
             * Parent component of chat messages, chat header and chat input
             * Input : Array of ReceivedMessage and SentMessage
             * Input : IChatViewOptions : Set style properties of ReceivedMessage, SentMessage, header, chat input, quick replies
             * Output: onOptionClicked event emitter : Triggered when user clicks one of the quick replies
             * Output: onInput event emitter : Triggered when user clicks of send button
             *
             * @export
             * @class ChatViewComponent
             * @implements {OnInit}
             */
            ,
            __metadata("design:paramtypes", [])
        ], ChatViewComponent);
        return ChatViewComponent;
    }());

    /*
    * Defines Message categories
    */
    var MESSAGE_CATEGORY = {
        SENT: 'sent',
        RECEIVED: 'received'
    };
    /*
    * Defines Message types
    */
    var MESSAGE_TYPES = {
        TEXT: 'text',
        IMAGE: 'image'
    };
    /**
     * Parent to: ReceivedMessage, SentMessage
     *
     * @export
     * @class Message
     */
    var Message = /** @class */ (function () {
        /**
         * Creates an instance of Message.
         * @param {*} category
         * @memberof Message
         */
        function Message(category) {
            this.category = category;
        }
        /**
         * Set text of message : Used for both Sent and Received Messages
         * @param {string} text
         * @returns Message
         * @memberof Message
         */
        Message.prototype.setText = function (text) {
            this.text = text;
            return this;
        };
        /**
         * Returns message text
         * @returns string
         * @memberof Message
         */
        Message.prototype.getText = function () {
            return this.text;
        };
        /**
         * Set image URL of message : Used for both Sent and Received Messages
         * @param {string} imageUrl
         * @returns Message
         * @memberof Message
         */
        Message.prototype.setImageUrl = function (imageUrl) {
            this.imageUrl = imageUrl;
            return this;
        };
        /**
         * Returns image URL
         * @returns string
         * @memberof Message
         */
        Message.prototype.getImageUrl = function () {
            return this.imageUrl;
        };
        /**
         * Returns message category
         * @returns string
         * @memberof Message
         */
        Message.prototype.getCategory = function () {
            return this.category;
        };
        return Message;
    }());
    /**
     * This class is used to create an instance of ReceivedMessage
     * It extends parent Message class with methods to set/get QuickReplies
     *
     * @export
     * @class ReceivedMessage
     * @extends {Message}
     */
    var ReceivedMessage = /** @class */ (function (_super) {
        __extends(ReceivedMessage, _super);
        /**
         *Creates an instance of ReceivedMessage.
         * @memberof ReceivedMessage
         */
        function ReceivedMessage() {
            /*
            * Set Message category as "Received" to parent class "Message"
            */
            return _super.call(this, MESSAGE_CATEGORY.RECEIVED) || this;
        }
        /**
         * Set quick replies of Receiveed Message
         * @param {string[]} quickReplies
         * @returns ReceivedMessage
         * @memberof ReceivedMessage
         */
        ReceivedMessage.prototype.setQuickReplies = function (quickReplies) {
            this.quickReplies = quickReplies;
            return this;
        };
        /**
         * Returns the set quick replies
         * @returns string[]
         * @memberof ReceivedMessage
         */
        ReceivedMessage.prototype.getQuickReplies = function () {
            return this.quickReplies;
        };
        return ReceivedMessage;
    }(Message));
    /**
     * This class is used to create an instance of SentMessage
     * It extends parent Message class no additional methods.
     *
     * @export
     * @class SentMessage
     * @extends {Message}
     */
    var SentMessage = /** @class */ (function (_super) {
        __extends(SentMessage, _super);
        /**
         *Creates an instance of SentMessage.
         * @memberof SentMessage
         */
        function SentMessage() {
            /*
           * Set Message category as "Sent" to parent class "Message"
           */
            return _super.call(this, MESSAGE_CATEGORY.SENT) || this;
        }
        return SentMessage;
    }(Message));

    var ReceivedMessageComponent = /** @class */ (function () {
        /**
         *Creates an instance of ReceivedMessageComponent.
         * @memberof ReceivedMessageComponent
         */
        function ReceivedMessageComponent() {
            /**
             * Emit an event to Parent on quick replies option is clicked
             *
             * @memberof ReceivedMessageComponent
             */
            this.onOptionClicked = new core.EventEmitter();
        }
        /**
         * @memberof ReceivedMessageComponent
         */
        ReceivedMessageComponent.prototype.ngOnInit = function () {
        };
        /**
         * This function act as a transporter and sends the received event from Message Component to Parent component
         * @param {string} option
         * @memberof ReceivedMessageComponent
         */
        ReceivedMessageComponent.prototype.optionClicked = function (option) {
            this.onOptionClicked.emit(option);
        };
        __decorate([
            core.Input('message'),
            __metadata("design:type", ReceivedMessage)
        ], ReceivedMessageComponent.prototype, "message", void 0);
        __decorate([
            core.Input('receivedMessageOptions'),
            __metadata("design:type", Object)
        ], ReceivedMessageComponent.prototype, "receivedMessageOptions", void 0);
        __decorate([
            core.Input('quickRepliesOptions'),
            __metadata("design:type", Object)
        ], ReceivedMessageComponent.prototype, "quickRepliesOptions", void 0);
        __decorate([
            core.Output('onOptionClicked'),
            __metadata("design:type", Object)
        ], ReceivedMessageComponent.prototype, "onOptionClicked", void 0);
        ReceivedMessageComponent = __decorate([
            core.Component({
                selector: 'received-message',
                template: "<div class=\"recv-msg\" >\n    <div *ngIf=\"message\" class=\"recv-wrapper\" >\n        <message [data]=\"message\" (onOptionClicked)=\"optionClicked($event)\" \n        [style.backgroundColor]=\"receivedMessageOptions?.backgroundColor\" \n        [receivedMessageOptions]=\"receivedMessageOptions\"\n        [quickRepliesOptions]=\"quickRepliesOptions\"\n        ></message>\n    </div>\n</div>",
                styles: ["div{color:#fff!important;text-align:left}.recv-msg{display:block;width:100%}.recv-msg .recv-wrapper{display:-webkit-box;display:flex;-webkit-box-pack:start;justify-content:flex-start;margin-right:20%}.recv-msg message{background-color:#00bcd4;padding:5px;border-radius:10px;margin-left:10px;max-width:90%}"]
            })
            /**
             * ReceivedMessageComponent class. This class wraps Message component for Received Message
             * Inputs: ReceivedMessage, IChatMessageOption, quickRepliesOptions
             * Outpus: onOptionClicked Event
             * @export
             * @class ReceivedMessageComponent
             * @implements {OnInit}
             */
            ,
            __metadata("design:paramtypes", [])
        ], ReceivedMessageComponent);
        return ReceivedMessageComponent;
    }());

    var SentMessageComponent = /** @class */ (function () {
        function SentMessageComponent() {
        }
        SentMessageComponent.prototype.ngOnInit = function () { };
        __decorate([
            core.Input('message'),
            __metadata("design:type", SentMessage)
        ], SentMessageComponent.prototype, "message", void 0);
        __decorate([
            core.Input('sentMessageOptions'),
            __metadata("design:type", Object)
        ], SentMessageComponent.prototype, "sentMessageOptions", void 0);
        SentMessageComponent = __decorate([
            core.Component({
                selector: 'sent-message',
                template: "<div class=\"sent-msg\">\n    <div *ngIf=\"message\" class=\"sent-wrapper\">\n        <message [data]=\"message\" [style.backgroundColor]=\"sentMessageOptions?.backgroundColor\" [sentMessageOptions]=\"sentMessageOptions\"></message>\n    </div>\n</div>",
                styles: ["div{color:#0c0c0c;text-align:left}.sent-msg{display:block;width:100%}.sent-msg .sent-wrapper{display:-webkit-box;display:flex;-webkit-box-pack:end;justify-content:flex-end;margin-left:20%}.sent-msg message{background-color:#bbdefb;padding:5px;border-radius:10px;margin-right:10px;max-width:90%}"]
            })
            /**
             * SentMessageComponent class. This class wraps Message component for Sent Message
             * Inputs: SentMessage, sentMessageOptions
             * @export
             * @class SentMessageComponent
             * @implements {OnInit}
             */
            ,
            __metadata("design:paramtypes", [])
        ], SentMessageComponent);
        return SentMessageComponent;
    }());

    var MessageComponent = /** @class */ (function () {
        /**
         *Creates an instance of MessageComponent.
         * @param {DomSanitizer} sanitizer
         * @memberof MessageComponent
         */
        function MessageComponent(sanitizer) {
            this.sanitizer = sanitizer;
            /**
             * Output property which will emit an event on quick reply option is clicked.
             * @memberof MessageComponent
             */
            this.onOptionClicked = new core.EventEmitter();
            this.isRecivedMessage = false;
            this.quickReplies = [];
        }
        /**
         * Set 'isRecivedMessage' flag based on message category.
         * Assign values to text, image and quick replies variables.
         * Currently image and quick replies are only used for received message.
         * @memberof MessageComponent
         */
        MessageComponent.prototype.ngOnInit = function () {
            if (!this.message) {
                throw new Error('Data Required');
            }
            if (this.message.getCategory() == MESSAGE_CATEGORY.RECEIVED) {
                this.isRecivedMessage = true;
            }
            this.text = this.message.getText();
            this.image = this.message.getImageUrl();
            if (this.message instanceof ReceivedMessage) {
                this.quickReplies = this.message.getQuickReplies();
            }
        };
        /**
         * Emit 'onOptionClicked' event with 'option text' payload
         * @param {string} option
         * @memberof MessageComponent
         */
        MessageComponent.prototype.optionClicked = function (option) {
            this.onOptionClicked.emit(option);
        };
        /**
         * Set mouseover HostBinding for message component
         * This sets BoxShadow of component on 'hover'
         * @memberof MessageComponent
         */
        MessageComponent.prototype.onMouseOver = function () {
            if (this.isRecivedMessage) {
                this.boxShadow = this.sanitizer.bypassSecurityTrustStyle('1px 2px 8px rgba(33,33,33,.4)');
            }
            else {
                this.boxShadow = this.sanitizer.bypassSecurityTrustStyle('-2px 2px 8px rgba(33,33,33,.4)');
            }
        };
        /**
        * Set mouseleave HostBinding for message component
        * This removes BoxShadow of component.
        * @memberof MessageComponent
        */
        MessageComponent.prototype.onMouseLeave = function () {
            this.boxShadow = 'none';
        };
        MessageComponent.ctorParameters = function () { return [
            { type: platformBrowser.DomSanitizer }
        ]; };
        __decorate([
            core.Input('data'),
            __metadata("design:type", Message)
        ], MessageComponent.prototype, "message", void 0);
        __decorate([
            core.Input('receivedMessageOptions'),
            __metadata("design:type", Object)
        ], MessageComponent.prototype, "receivedMessageOptions", void 0);
        __decorate([
            core.Input('sentMessageOptions'),
            __metadata("design:type", Object)
        ], MessageComponent.prototype, "sentMessageOptions", void 0);
        __decorate([
            core.Input('quickRepliesOptions'),
            __metadata("design:type", Object)
        ], MessageComponent.prototype, "quickRepliesOptions", void 0);
        __decorate([
            core.Output('onOptionClicked'),
            __metadata("design:type", Object)
        ], MessageComponent.prototype, "onOptionClicked", void 0);
        __decorate([
            core.HostBinding('style.boxShadow'),
            __metadata("design:type", Object)
        ], MessageComponent.prototype, "boxShadow", void 0);
        __decorate([
            core.HostListener('mouseover'),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", []),
            __metadata("design:returntype", void 0)
        ], MessageComponent.prototype, "onMouseOver", null);
        __decorate([
            core.HostListener('mouseleave'),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", []),
            __metadata("design:returntype", void 0)
        ], MessageComponent.prototype, "onMouseLeave", null);
        MessageComponent = __decorate([
            core.Component({
                selector: 'message',
                template: "<div *ngIf=\"message\">\n    <!--RECEIVED ITEM-->\n    <div class=\"message-wrapper\" *ngIf=\"isRecivedMessage\">\n\n        <div class=\"recv-text\" *ngIf=\"text\" [style.color]=\"receivedMessageOptions?.textColor\">{{text}}</div>\n\n        <div *ngIf=\"image\">\n            <img [src]=\"image\" alt=\"\">\n        </div>\n        <div class=\"quickreply-wrapper\" *ngIf=\"quickReplies && quickReplies.length>0\">\n            <div class=\"recv-quickreply\" *ngFor=\"let reply of quickReplies\"\n                [style.backgroundColor]=\"quickRepliesOptions?.backgroundColor\"\n                [style.borderColor]=\"quickRepliesOptions?.borderColor\">\n                <div (click)=\"optionClicked(reply)\" \n                [style.color]=\"quickRepliesOptions?.textColor\">{{reply}}</div>\n            </div>\n        </div>\n    </div>\n\n    <!--SENT ITEM-->\n    <div class=\"message-wrapper\" *ngIf=\"!isRecivedMessage\">\n        <div class=\"sent-text\" *ngIf=\"text\" [style.color]=\"sentMessageOptions?.textColor\">{{text}}</div>\n\n        <div *ngIf=\"image\">\n            <img [src]=\"image\" alt=\"\">\n        </div>\n    </div>\n</div>",
                styles: ["img{width:100%;display:block;margin:auto;max-width:95%;padding:5px;border-radius:10px}div{font-size:.9rem}.recv-text,.sent-text{word-wrap:break-word;padding:3px}.recv-quickreply{display:inline-block;border:1px solid rgba(255,255,255,.521);border-radius:10px;padding:3px 5px;margin-left:3px;cursor:pointer}.recv-quickreply:hover{box-shadow:0 0 11px rgba(33,33,33,.2);border:1px solid #fff}.quickreply-wrapper{margin-top:5px}"]
            }),
            __metadata("design:paramtypes", [platformBrowser.DomSanitizer])
        ], MessageComponent);
        return MessageComponent;
    }());

    var UserInputComponent = /** @class */ (function () {
        function UserInputComponent() {
            /**
             * Event emitter for send button click
             * @memberof UserInputComponent
             */
            this.onInput = new core.EventEmitter();
        }
        UserInputComponent.prototype.ngOnInit = function () {
        };
        /**
         * Triggered when send button is clicked
         * Triggers OnInput event with input text as data if input text is not null
         * Clears the input text on event emit
         * @memberof UserInputComponent
         */
        UserInputComponent.prototype.send = function () {
            if (this.inputText) {
                this.onInput.emit(this.inputText);
                this.inputText = "";
            }
        };
        __decorate([
            core.Output('onInput'),
            __metadata("design:type", Object)
        ], UserInputComponent.prototype, "onInput", void 0);
        __decorate([
            core.Input('options'),
            __metadata("design:type", Object)
        ], UserInputComponent.prototype, "options", void 0);
        UserInputComponent = __decorate([
            core.Component({
                selector: 'user-input',
                template: "<div class=\"input-wrapper\" [style.backgroundColor]=\"options?.backgroundColor\">\n    <input type=\"text\" [(ngModel)]=\"inputText\" placeholder=\"Say Something...\" \n    [attr.maxlength]=\"options?.maxLength\"\n    [style.borderColor]=\"options?.borderColor\"\n    [style.color]=\"options?.textColor\" (keyup.enter)=\"send()\"/>\n    <button class=\"btn-input-send\"  (click)=\"send()\">\n        <send-svg  [color]=\"options?.sendIconColor\"></send-svg>\n    </button>\n</div>",
                styles: [".input-wrapper{display:-webkit-box;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;flex-direction:row;padding-left:5px;height:35px;background-color:transparent}input{box-sizing:border-box;-webkit-box-flex:1;flex:1;border:1px solid #2196f3;background:0 0;color:#000;padding-left:5px;padding-right:5px;margin:2px;border-radius:15px}.btn-input-send{-webkit-box-flex:0;flex:0}button{background:0 0;border:none}::-webkit-input-placeholder{color:gray}:-ms-input-placeholder{color:gray}::-moz-placeholder{color:gray;opacity:.5}::-ms-input-placeholder{color:gray;opacity:.5}::placeholder{color:gray;opacity:.5}button:focus,input:focus,select:focus,textarea:focus{outline:0}send-svg{display:block;height:20px;width:20px;padding:5px 0;-webkit-transition:ease-in;transition:ease-in}send-svg:hover{height:18px;width:18px}"]
            })
            /**
             * UserInputComponent class. This class wraps user message input box and Send SVG Component.
             * Inputs: options- Set styling options for UserInputComponent
             * @export
             * @class UserInputComponent
             * @implements {OnInit}
             */
            ,
            __metadata("design:paramtypes", [])
        ], UserInputComponent);
        return UserInputComponent;
    }());

    var ChatHeaderComponent = /** @class */ (function () {
        /**
         *Creates an instance of ChatHeaderComponent.
         * @memberof ChatHeaderComponent
         */
        function ChatHeaderComponent() {
            this.title = 'Chat View';
        }
        ChatHeaderComponent.prototype.ngOnInit = function () {
        };
        __decorate([
            core.Input('title'),
            __metadata("design:type", Object)
        ], ChatHeaderComponent.prototype, "title", void 0);
        __decorate([
            core.Input('headerOptions'),
            __metadata("design:type", Object)
        ], ChatHeaderComponent.prototype, "headerOptions", void 0);
        ChatHeaderComponent = __decorate([
            core.Component({
                selector: 'chat-header',
                template: "<div class=\"chat-header-wrapper\" [style.backgroundColor]=\"headerOptions?.backgroundColor\">\n    <div class=\"chat-header-title\" [style.color]=\"headerOptions?.textColor\">\n        {{title}}\n    </div>\n</div>",
                styles: [".chat-header-wrapper{background-color:#2196f3;height:40px;display:-webkit-box;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;justify-content:space-around}.chat-header-title{text-align:center;vertical-align:middle;color:#fff}"]
            }),
            __metadata("design:paramtypes", [])
        ], ChatHeaderComponent);
        return ChatHeaderComponent;
    }());

    var SendSVGImage = /** @class */ (function () {
        function SendSVGImage() {
            this.color = "";
        }
        SendSVGImage.prototype.ngOnInit = function () {
        };
        __decorate([
            core.Input('color'),
            __metadata("design:type", Object)
        ], SendSVGImage.prototype, "color", void 0);
        SendSVGImage = __decorate([
            core.Component({
                selector: 'send-svg',
                template: "<svg enable-background=\"new 0 0 24 24\" height=\"100%\" viewBox=\"0 0 24 24\" width=\"100%\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"m8.75 17.612v4.638c0 .324.208.611.516.713.077.025.156.037.234.037.234 0 .46-.11.604-.306l2.713-3.692z\" [attr.fill]=\"color\"/><path d=\"m23.685.139c-.23-.163-.532-.185-.782-.054l-22.5 11.75c-.266.139-.423.423-.401.722.023.3.222.556.505.653l6.255 2.138 13.321-11.39-10.308 12.419 10.483 3.583c.078.026.16.04.242.04.136 0 .271-.037.39-.109.19-.116.319-.311.352-.53l2.75-18.5c.041-.28-.077-.558-.307-.722z\" [attr.fill]=\"color\"/></svg>",
                styles: [""]
            })
            /**
             * SVG Image component for send icon image.
             * Color of this SVG component can be set dynamically using 'color' input property
             * @export
             * @class SendSVGImage
             * @implements {OnInit}
             */
            ,
            __metadata("design:paramtypes", [])
        ], SendSVGImage);
        return SendSVGImage;
    }());

    var ChatViewModule = /** @class */ (function () {
        function ChatViewModule() {
        }
        ChatViewModule = __decorate([
            core.NgModule({
                imports: [
                    common.CommonModule,
                    forms.FormsModule
                ],
                declarations: [ChatViewComponent,
                    ReceivedMessageComponent,
                    SentMessageComponent,
                    MessageComponent,
                    UserInputComponent,
                    ChatHeaderComponent,
                    SendSVGImage
                ],
                exports: [ChatViewComponent]
            })
        ], ChatViewModule);
        return ChatViewModule;
    }());

    exports.ChatViewComponent = ChatViewComponent;
    exports.ChatViewModule = ChatViewModule;
    exports.ChatViewService = ChatViewService;
    exports.MESSAGE_CATEGORY = MESSAGE_CATEGORY;
    exports.MESSAGE_TYPES = MESSAGE_TYPES;
    exports.Message = Message;
    exports.ReceivedMessage = ReceivedMessage;
    exports.SentMessage = SentMessage;
    exports.ɵa = messagePushAnimation;
    exports.ɵb = ReceivedMessageComponent;
    exports.ɵc = SentMessageComponent;
    exports.ɵd = MessageComponent;
    exports.ɵe = UserInputComponent;
    exports.ɵf = ChatHeaderComponent;
    exports.ɵg = SendSVGImage;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
