// chat-view/esm5/lib/components/message-data/message-data.js
import * as tslib_1 from "tslib";
/*
* Defines Message categories
*/
export var MESSAGE_CATEGORY = {
    SENT: 'sent',
    RECEIVED: 'received'
};
/*
* Defines Message types
*/
export var MESSAGE_TYPES = {
    TEXT: 'text',
    IMAGE: 'image'
};
/**
 * Parent to: ReceivedMessage, SentMessage
 *
 * @export
 * @class Message
 */
var Message = /** @class */ (function () {
    /**
     * Creates an instance of Message.
     * @param {*} category
     * @memberof Message
     */
    function Message(category) {
        this.category = category;
    }
    /**
     * Set text of message : Used for both Sent and Received Messages
     * @param {string} text
     * @returns Message
     * @memberof Message
     */
    Message.prototype.setText = function (text) {
        this.text = text;
        return this;
    };
    /**
     * Returns message text
     * @returns string
     * @memberof Message
     */
    Message.prototype.getText = function () {
        return this.text;
    };
    /**
     * Set image URL of message : Used for both Sent and Received Messages
     * @param {string} imageUrl
     * @returns Message
     * @memberof Message
     */
    Message.prototype.setImageUrl = function (imageUrl) {
        this.imageUrl = imageUrl;
        return this;
    };
    /**
     * Returns image URL
     * @returns string
     * @memberof Message
     */
    Message.prototype.getImageUrl = function () {
        return this.imageUrl;
    };
    /**
     * Returns message category
     * @returns string
     * @memberof Message
     */
    Message.prototype.getCategory = function () {
        return this.category;
    };
    return Message;
}());
export { Message };
/**
 * This class is used to create an instance of ReceivedMessage
 * It extends parent Message class with methods to set/get QuickReplies
 *
 * @export
 * @class ReceivedMessage
 * @extends {Message}
 */
var ReceivedMessage = /** @class */ (function (_super) {
    tslib_1.__extends(ReceivedMessage, _super);
    /**
     *Creates an instance of ReceivedMessage.
     * @memberof ReceivedMessage
     */
    function ReceivedMessage() {
        /*
        * Set Message category as "Received" to parent class "Message"
        */
        return _super.call(this, MESSAGE_CATEGORY.RECEIVED) || this;
    }
    /**
     * Set quick replies of Receiveed Message
     * @param {string[]} quickReplies
     * @returns ReceivedMessage
     * @memberof ReceivedMessage
     */
    ReceivedMessage.prototype.setQuickReplies = function (quickReplies) {
        this.quickReplies = quickReplies;
        return this;
    };
    /**
     * Returns the set quick replies
     * @returns string[]
     * @memberof ReceivedMessage
     */
    ReceivedMessage.prototype.getQuickReplies = function () {
        return this.quickReplies;
    };
    return ReceivedMessage;
}(Message));
export { ReceivedMessage };
/**
 * This class is used to create an instance of SentMessage
 * It extends parent Message class no additional methods.
 *
 * @export
 * @class SentMessage
 * @extends {Message}
 */
var SentMessage = /** @class */ (function (_super) {
    tslib_1.__extends(SentMessage, _super);
    /**
     *Creates an instance of SentMessage.
     * @memberof SentMessage
     */
    function SentMessage() {
        /*
       * Set Message category as "Sent" to parent class "Message"
       */
        return _super.call(this, MESSAGE_CATEGORY.SENT) || this;
    }
    return SentMessage;
}(Message));
export { SentMessage };
