// chat-view/esm2015/lib/components/received-msg/received-msg.component.js
import * as tslib_1 from "tslib";
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { ReceivedMessage } from '../message-data/message-data';
let ReceivedMessageComponent = 
/**
 * ReceivedMessageComponent class. This class wraps Message component for Received Message
 * Inputs: ReceivedMessage, IChatMessageOption, quickRepliesOptions
 * Outpus: onOptionClicked Event
 * @export
 * @class ReceivedMessageComponent
 * @implements {OnInit}
 */
class ReceivedMessageComponent {
    /**
     *Creates an instance of ReceivedMessageComponent.
     * @memberof ReceivedMessageComponent
     */
    constructor() {
        /**
         * Emit an event to Parent on quick replies option is clicked
         *
         * @memberof ReceivedMessageComponent
         */
        this.onOptionClicked = new EventEmitter();
    }
    /**
     * @memberof ReceivedMessageComponent
     */
    ngOnInit() {
    }
    /**
     * This function act as a transporter and sends the received event from Message Component to Parent component
     * @param {string} option
     * @memberof ReceivedMessageComponent
     */
    optionClicked(option) {
        this.onOptionClicked.emit(option);
    }
};
tslib_1.__decorate([
    Input('message'),
    tslib_1.__metadata("design:type", ReceivedMessage)
], ReceivedMessageComponent.prototype, "message", void 0);
tslib_1.__decorate([
    Input('receivedMessageOptions'),
    tslib_1.__metadata("design:type", Object)
], ReceivedMessageComponent.prototype, "receivedMessageOptions", void 0);
tslib_1.__decorate([
    Input('quickRepliesOptions'),
    tslib_1.__metadata("design:type", Object)
], ReceivedMessageComponent.prototype, "quickRepliesOptions", void 0);
tslib_1.__decorate([
    Output('onOptionClicked'),
    tslib_1.__metadata("design:type", Object)
], ReceivedMessageComponent.prototype, "onOptionClicked", void 0);
ReceivedMessageComponent = tslib_1.__decorate([
    Component({
        selector: 'received-message',
        template: "<div class=\"recv-msg\" >\n    <div *ngIf=\"message\" class=\"recv-wrapper\" >\n        <message [data]=\"message\" (onOptionClicked)=\"optionClicked($event)\" \n        [style.backgroundColor]=\"receivedMessageOptions?.backgroundColor\" \n        [receivedMessageOptions]=\"receivedMessageOptions\"\n        [quickRepliesOptions]=\"quickRepliesOptions\"\n        ></message>\n    </div>\n</div>",
        styles: ["div{color:#fff!important;text-align:left}.recv-msg{display:block;width:100%}.recv-msg .recv-wrapper{display:-webkit-box;display:flex;-webkit-box-pack:start;justify-content:flex-start;margin-right:20%}.recv-msg message{background-color:#00bcd4;padding:5px;border-radius:10px;margin-left:10px;max-width:90%}"]
    })
    /**
     * ReceivedMessageComponent class. This class wraps Message component for Received Message
     * Inputs: ReceivedMessage, IChatMessageOption, quickRepliesOptions
     * Outpus: onOptionClicked Event
     * @export
     * @class ReceivedMessageComponent
     * @implements {OnInit}
     */
    ,
    tslib_1.__metadata("design:paramtypes", [])
], ReceivedMessageComponent);
export { ReceivedMessageComponent };
