// chat-view/fesm2015/chat-view.js
import { __decorate, __metadata } from 'tslib';
import { ɵɵdefineInjectable, Injectable, EventEmitter, ViewChild, ElementRef, Input, Output, Component, HostBinding, HostListener, NgModule } from '@angular/core';
import { animation, style, animate, trigger, transition, query, stagger, useAnimation } from '@angular/animations';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { DomSanitizer } from '@angular/platform-browser';

let ChatViewService = class ChatViewService {
    constructor() { }
};
ChatViewService.ngInjectableDef = ɵɵdefineInjectable({ factory: function ChatViewService_Factory() { return new ChatViewService(); }, token: ChatViewService, providedIn: "root" });
ChatViewService = __decorate([
    Injectable({
        providedIn: 'root'
    }),
    __metadata("design:paramtypes", [])
], ChatViewService);

const messagePushAnimation = animation([
    style({
        opacity: '{{ opacity }}',
        transform: 'translateY({{translateY}}%)'
    }),
    animate('{{ time }}')
]);

let ChatViewComponent = 
/**
 * This is only componet that needs to be added while integrating chat-view library.
 * Parent component of chat messages, chat header and chat input
 * Input : Array of ReceivedMessage and SentMessage
 * Input : IChatViewOptions : Set style properties of ReceivedMessage, SentMessage, header, chat input, quick replies
 * Output: onOptionClicked event emitter : Triggered when user clicks one of the quick replies
 * Output: onInput event emitter : Triggered when user clicks of send button
 *
 * @export
 * @class ChatViewComponent
 * @implements {OnInit}
 */
class ChatViewComponent {
    constructor() {
        /**
         * Input messages (recieved or sent) array passed to component.
         * @type {(ReceivedMessage[] | SentMessage[])}
         * @memberof ChatViewComponent
         */
        this.messages = [];
        /**
         * Input property to set styling options to chat-view compoenents
         * @type {IChatViewOptions}
         * @memberof ChatViewComponent
         */
        this.options = {
            //Set background color and text color of SentMessage
            sentMessage: {
                textColor: 'white',
                backgroundColor: '#c7ac7b',
            },
            //Set background color and text color of ReceivedMessage
            receivedMessage: {
                textColor: 'black',
                backgroundColor: 'white',
            },
            //Set background color and text color of ChatHeader
            header: {
                textColor: 'white',
                backgroundColor: '#71a7de'
            },
            //Set border color, text color, max input length, send icon color and background color of input box
            inputBox: {
                borderColor: 'gray',
                textColor: 'black',
                maxLength: 50,
                sendIconColor: '#5e8dbd',
                backgroundColor: '#e1faf8'
            },
            //Set text color, background color and border color of quick replies
            quickReplies: {
                textColor: 'white',
                backgroundColor: '#c7ac7b',
                borderColor: '#c7ac7b'
            },
            //Set background color of chatlist pane
            chatlistBackgroundColor: '#e1faf8',
        };
        /**
         * Event emitter for quick replies are clicked
         * @memberof ChatViewComponent
         */
        this.onOptionClicked = new EventEmitter();
        /**
         * Event emitter when send button is clicked
         * @memberof ChatViewComponent
         */
        this.onInput = new EventEmitter();
        //messages array to keep track of previous messages.
        //This is used in ngDoCheck() top scroll chatview if new message is added.
        this.oldMessages = [];
    }
    /**
     * Event emitter when send button is clicked.
     * Parent copmponent will recived users query. Then it can be added in sent message array.
     * @param {*} text
     * @memberof ChatViewComponent
     */
    sendMessage(text) {
        this.onInput.emit(text);
    }
    /**
     * Event emitter when quick replies are clicked.
     * Parent copmponent will recived selected option. Then it can be added in sent message array.
     * @param {*} text
     * @memberof ChatViewComponent
     */
    optionClicked(option) {
        this.onInput.emit(option);
    }
    /**
     * To scroll chatlist to the bottom
     * @memberof ChatViewComponent
     */
    scrollToBottom() {
        setTimeout(() => {
            this.chatlist.nativeElement.scrollTop = this.chatlist.nativeElement.scrollHeight;
        }, 0);
    }
    /**
     * After every change detection cycle, check if new message is added in message array.
     * If yes, then scroll chat list to bottom.
     * @memberof ChatViewComponent
     */
    ngDoCheck() {
        if (this.messages && this.messages.length != this.oldMessages.length) {
            this.scrollToBottom();
        }
        this.oldMessages = [...this.messages];
    }
};
__decorate([
    ViewChild('chatlist', { static: false }),
    __metadata("design:type", ElementRef
    /**
     * Input messages (recieved or sent) array passed to component.
     * @type {(ReceivedMessage[] | SentMessage[])}
     * @memberof ChatViewComponent
     */
    )
], ChatViewComponent.prototype, "chatlist", void 0);
__decorate([
    Input('messages'),
    __metadata("design:type", Array)
], ChatViewComponent.prototype, "messages", void 0);
__decorate([
    Input('options'),
    __metadata("design:type", Object)
], ChatViewComponent.prototype, "options", void 0);
__decorate([
    Output('onOptionClicked'),
    __metadata("design:type", Object)
], ChatViewComponent.prototype, "onOptionClicked", void 0);
__decorate([
    Output('onInput'),
    __metadata("design:type", Object)
], ChatViewComponent.prototype, "onInput", void 0);
ChatViewComponent = __decorate([
    Component({
        selector: 'chat-view',
        template: "<div class=\"chat-container\">\n    <chat-header title=\"My Chat View\" [headerOptions]=\"options.header\"></chat-header>\n    <div #chatlist class=\"chat-list-wrapper\"\n      [@messageAddedAnimation]=\"messages.length\"\n      [style.backgroundColor]=\"options.chatlistBackgroundColor\"\n      >\n        <div  class=\"chat-list-item\" *ngFor=\"let message of messages;let i=index\">\n\n            <!--received Message-->\n            <div *ngIf=\"message.getCategory()=='received'\">\n                <received-message [message]=\"message\"\n                (onOptionClicked)=\"optionClicked($event)\" \n                [receivedMessageOptions]=\"options.receivedMessage\"\n                [quickRepliesOptions]=\"options.quickReplies\"\n                ></received-message>\n            </div>\n\n            <!--Sent Message-->\n            <div *ngIf=\"message.getCategory()=='sent'\">\n                <sent-message [message]=\"message\"\n                [sentMessageOptions]=\"options.sentMessage\"\n                ></sent-message>\n            </div>\n        </div>\n    </div>\n    <user-input (onInput)=\"sendMessage($event)\"\n    [options]=\"options.inputBox\"></user-input>\n</div>",
        animations: [
            trigger('messageAddedAnimation', [
                transition(':enter, * => 0, * => -1', []),
                transition(':increment', [
                    query(':enter', [
                        style({ opacity: 0 }),
                        stagger(100, [
                            useAnimation(messagePushAnimation, {
                                params: {
                                    opacity: 1,
                                    translateY: 10,
                                    time: '0.2s'
                                }
                            })
                        ]),
                    ], { optional: true })
                ]),
            ]),
        ],
        styles: ["div{color:red}.chat-container{width:100%;height:100%;display:-webkit-box;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;-webkit-box-pack:end;justify-content:flex-end}.chat-list-wrapper{-webkit-box-flex:1;flex:1 1 auto;overflow-y:scroll}.chat-list-wrapper .chat-list-item:first-child{margin-top:5px}.chat-list-wrapper .chat-list-item,.chat-list-wrapper received-message sent-message{margin-bottom:5px}chat-header{box-shadow:0 5px 11px #dbd9d9be;z-index:1}user-input{-webkit-box-flex:0;flex:0 0 auto;background:0 0;z-index:1}received-message,sent-message{display:block;width:100%}"]
    })
    /**
     * This is only componet that needs to be added while integrating chat-view library.
     * Parent component of chat messages, chat header and chat input
     * Input : Array of ReceivedMessage and SentMessage
     * Input : IChatViewOptions : Set style properties of ReceivedMessage, SentMessage, header, chat input, quick replies
     * Output: onOptionClicked event emitter : Triggered when user clicks one of the quick replies
     * Output: onInput event emitter : Triggered when user clicks of send button
     *
     * @export
     * @class ChatViewComponent
     * @implements {OnInit}
     */
    ,
    __metadata("design:paramtypes", [])
], ChatViewComponent);

/*
* Defines Message categories
*/
const MESSAGE_CATEGORY = {
    SENT: 'sent',
    RECEIVED: 'received'
};
/*
* Defines Message types
*/
const MESSAGE_TYPES = {
    TEXT: 'text',
    IMAGE: 'image'
};
/**
 * Parent to: ReceivedMessage, SentMessage
 *
 * @export
 * @class Message
 */
class Message {
    /**
     * Creates an instance of Message.
     * @param {*} category
     * @memberof Message
     */
    constructor(category) {
        this.category = category;
    }
    /**
     * Set text of message : Used for both Sent and Received Messages
     * @param {string} text
     * @returns Message
     * @memberof Message
     */
    setText(text) {
        this.text = text;
        return this;
    }
    /**
     * Returns message text
     * @returns string
     * @memberof Message
     */
    getText() {
        return this.text;
    }
    /**
     * Set image URL of message : Used for both Sent and Received Messages
     * @param {string} imageUrl
     * @returns Message
     * @memberof Message
     */
    setImageUrl(imageUrl) {
        this.imageUrl = imageUrl;
        return this;
    }
    /**
     * Returns image URL
     * @returns string
     * @memberof Message
     */
    getImageUrl() {
        return this.imageUrl;
    }
    /**
     * Returns message category
     * @returns string
     * @memberof Message
     */
    getCategory() {
        return this.category;
    }
}
/**
 * This class is used to create an instance of ReceivedMessage
 * It extends parent Message class with methods to set/get QuickReplies
 *
 * @export
 * @class ReceivedMessage
 * @extends {Message}
 */
class ReceivedMessage extends Message {
    /**
     *Creates an instance of ReceivedMessage.
     * @memberof ReceivedMessage
     */
    constructor() {
        /*
        * Set Message category as "Received" to parent class "Message"
        */
        super(MESSAGE_CATEGORY.RECEIVED);
    }
    /**
     * Set quick replies of Receiveed Message
     * @param {string[]} quickReplies
     * @returns ReceivedMessage
     * @memberof ReceivedMessage
     */
    setQuickReplies(quickReplies) {
        this.quickReplies = quickReplies;
        return this;
    }
    /**
     * Returns the set quick replies
     * @returns string[]
     * @memberof ReceivedMessage
     */
    getQuickReplies() {
        return this.quickReplies;
    }
}
/**
 * This class is used to create an instance of SentMessage
 * It extends parent Message class no additional methods.
 *
 * @export
 * @class SentMessage
 * @extends {Message}
 */
class SentMessage extends Message {
    /**
     *Creates an instance of SentMessage.
     * @memberof SentMessage
     */
    constructor() {
        /*
       * Set Message category as "Sent" to parent class "Message"
       */
        super(MESSAGE_CATEGORY.SENT);
    }
}

let ReceivedMessageComponent = 
/**
 * ReceivedMessageComponent class. This class wraps Message component for Received Message
 * Inputs: ReceivedMessage, IChatMessageOption, quickRepliesOptions
 * Outpus: onOptionClicked Event
 * @export
 * @class ReceivedMessageComponent
 * @implements {OnInit}
 */
class ReceivedMessageComponent {
    /**
     *Creates an instance of ReceivedMessageComponent.
     * @memberof ReceivedMessageComponent
     */
    constructor() {
        /**
         * Emit an event to Parent on quick replies option is clicked
         *
         * @memberof ReceivedMessageComponent
         */
        this.onOptionClicked = new EventEmitter();
    }
    /**
     * @memberof ReceivedMessageComponent
     */
    ngOnInit() {
    }
    /**
     * This function act as a transporter and sends the received event from Message Component to Parent component
     * @param {string} option
     * @memberof ReceivedMessageComponent
     */
    optionClicked(option) {
        this.onOptionClicked.emit(option);
    }
};
__decorate([
    Input('message'),
    __metadata("design:type", ReceivedMessage)
], ReceivedMessageComponent.prototype, "message", void 0);
__decorate([
    Input('receivedMessageOptions'),
    __metadata("design:type", Object)
], ReceivedMessageComponent.prototype, "receivedMessageOptions", void 0);
__decorate([
    Input('quickRepliesOptions'),
    __metadata("design:type", Object)
], ReceivedMessageComponent.prototype, "quickRepliesOptions", void 0);
__decorate([
    Output('onOptionClicked'),
    __metadata("design:type", Object)
], ReceivedMessageComponent.prototype, "onOptionClicked", void 0);
ReceivedMessageComponent = __decorate([
    Component({
        selector: 'received-message',
        template: "<div class=\"recv-msg\" >\n    <div *ngIf=\"message\" class=\"recv-wrapper\" >\n        <message [data]=\"message\" (onOptionClicked)=\"optionClicked($event)\" \n        [style.backgroundColor]=\"receivedMessageOptions?.backgroundColor\" \n        [receivedMessageOptions]=\"receivedMessageOptions\"\n        [quickRepliesOptions]=\"quickRepliesOptions\"\n        ></message>\n    </div>\n</div>",
        styles: ["div{color:#fff!important;text-align:left}.recv-msg{display:block;width:100%}.recv-msg .recv-wrapper{display:-webkit-box;display:flex;-webkit-box-pack:start;justify-content:flex-start;margin-right:20%}.recv-msg message{background-color:#00bcd4;padding:5px;border-radius:10px;margin-left:10px;max-width:90%}"]
    })
    /**
     * ReceivedMessageComponent class. This class wraps Message component for Received Message
     * Inputs: ReceivedMessage, IChatMessageOption, quickRepliesOptions
     * Outpus: onOptionClicked Event
     * @export
     * @class ReceivedMessageComponent
     * @implements {OnInit}
     */
    ,
    __metadata("design:paramtypes", [])
], ReceivedMessageComponent);

let SentMessageComponent = 
/**
 * SentMessageComponent class. This class wraps Message component for Sent Message
 * Inputs: SentMessage, sentMessageOptions
 * @export
 * @class SentMessageComponent
 * @implements {OnInit}
 */
class SentMessageComponent {
    constructor() { }
    ngOnInit() { }
};
__decorate([
    Input('message'),
    __metadata("design:type", SentMessage)
], SentMessageComponent.prototype, "message", void 0);
__decorate([
    Input('sentMessageOptions'),
    __metadata("design:type", Object)
], SentMessageComponent.prototype, "sentMessageOptions", void 0);
SentMessageComponent = __decorate([
    Component({
        selector: 'sent-message',
        template: "<div class=\"sent-msg\">\n    <div *ngIf=\"message\" class=\"sent-wrapper\">\n        <message [data]=\"message\" [style.backgroundColor]=\"sentMessageOptions?.backgroundColor\" [sentMessageOptions]=\"sentMessageOptions\"></message>\n    </div>\n</div>",
        styles: ["div{color:#0c0c0c;text-align:left}.sent-msg{display:block;width:100%}.sent-msg .sent-wrapper{display:-webkit-box;display:flex;-webkit-box-pack:end;justify-content:flex-end;margin-left:20%}.sent-msg message{background-color:#bbdefb;padding:5px;border-radius:10px;margin-right:10px;max-width:90%}"]
    })
    /**
     * SentMessageComponent class. This class wraps Message component for Sent Message
     * Inputs: SentMessage, sentMessageOptions
     * @export
     * @class SentMessageComponent
     * @implements {OnInit}
     */
    ,
    __metadata("design:paramtypes", [])
], SentMessageComponent);

let MessageComponent = class MessageComponent {
    /**
     *Creates an instance of MessageComponent.
     * @param {DomSanitizer} sanitizer
     * @memberof MessageComponent
     */
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
        /**
         * Output property which will emit an event on quick reply option is clicked.
         * @memberof MessageComponent
         */
        this.onOptionClicked = new EventEmitter();
        this.isRecivedMessage = false;
        this.quickReplies = [];
    }
    /**
     * Set 'isRecivedMessage' flag based on message category.
     * Assign values to text, image and quick replies variables.
     * Currently image and quick replies are only used for received message.
     * @memberof MessageComponent
     */
    ngOnInit() {
        if (!this.message) {
            throw new Error('Data Required');
        }
        if (this.message.getCategory() == MESSAGE_CATEGORY.RECEIVED) {
            this.isRecivedMessage = true;
        }
        this.text = this.message.getText();
        this.image = this.message.getImageUrl();
        if (this.message instanceof ReceivedMessage) {
            this.quickReplies = this.message.getQuickReplies();
        }
    }
    /**
     * Emit 'onOptionClicked' event with 'option text' payload
     * @param {string} option
     * @memberof MessageComponent
     */
    optionClicked(option) {
        this.onOptionClicked.emit(option);
    }
    /**
     * Set mouseover HostBinding for message component
     * This sets BoxShadow of component on 'hover'
     * @memberof MessageComponent
     */
    onMouseOver() {
        if (this.isRecivedMessage) {
            this.boxShadow = this.sanitizer.bypassSecurityTrustStyle('1px 2px 8px rgba(33,33,33,.4)');
        }
        else {
            this.boxShadow = this.sanitizer.bypassSecurityTrustStyle('-2px 2px 8px rgba(33,33,33,.4)');
        }
    }
    /**
    * Set mouseleave HostBinding for message component
    * This removes BoxShadow of component.
    * @memberof MessageComponent
    */
    onMouseLeave() {
        this.boxShadow = 'none';
    }
};
MessageComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
__decorate([
    Input('data'),
    __metadata("design:type", Message)
], MessageComponent.prototype, "message", void 0);
__decorate([
    Input('receivedMessageOptions'),
    __metadata("design:type", Object)
], MessageComponent.prototype, "receivedMessageOptions", void 0);
__decorate([
    Input('sentMessageOptions'),
    __metadata("design:type", Object)
], MessageComponent.prototype, "sentMessageOptions", void 0);
__decorate([
    Input('quickRepliesOptions'),
    __metadata("design:type", Object)
], MessageComponent.prototype, "quickRepliesOptions", void 0);
__decorate([
    Output('onOptionClicked'),
    __metadata("design:type", Object)
], MessageComponent.prototype, "onOptionClicked", void 0);
__decorate([
    HostBinding('style.boxShadow'),
    __metadata("design:type", Object)
], MessageComponent.prototype, "boxShadow", void 0);
__decorate([
    HostListener('mouseover'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], MessageComponent.prototype, "onMouseOver", null);
__decorate([
    HostListener('mouseleave'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], MessageComponent.prototype, "onMouseLeave", null);
MessageComponent = __decorate([
    Component({
        selector: 'message',
        template: "<div *ngIf=\"message\">\n    <!--RECEIVED ITEM-->\n    <div class=\"message-wrapper\" *ngIf=\"isRecivedMessage\">\n\n        <div class=\"recv-text\" *ngIf=\"text\" [style.color]=\"receivedMessageOptions?.textColor\">{{text}}</div>\n\n        <div *ngIf=\"image\">\n            <img [src]=\"image\" alt=\"\">\n        </div>\n        <div class=\"quickreply-wrapper\" *ngIf=\"quickReplies && quickReplies.length>0\">\n            <div class=\"recv-quickreply\" *ngFor=\"let reply of quickReplies\"\n                [style.backgroundColor]=\"quickRepliesOptions?.backgroundColor\"\n                [style.borderColor]=\"quickRepliesOptions?.borderColor\">\n                <div (click)=\"optionClicked(reply)\" \n                [style.color]=\"quickRepliesOptions?.textColor\">{{reply}}</div>\n            </div>\n        </div>\n    </div>\n\n    <!--SENT ITEM-->\n    <div class=\"message-wrapper\" *ngIf=\"!isRecivedMessage\">\n        <div class=\"sent-text\" *ngIf=\"text\" [style.color]=\"sentMessageOptions?.textColor\">{{text}}</div>\n\n        <div *ngIf=\"image\">\n            <img [src]=\"image\" alt=\"\">\n        </div>\n    </div>\n</div>",
        styles: ["img{width:100%;display:block;margin:auto;max-width:95%;padding:5px;border-radius:10px}div{font-size:.9rem}.recv-text,.sent-text{word-wrap:break-word;padding:3px}.recv-quickreply{display:inline-block;border:1px solid rgba(255,255,255,.521);border-radius:10px;padding:3px 5px;margin-left:3px;cursor:pointer}.recv-quickreply:hover{box-shadow:0 0 11px rgba(33,33,33,.2);border:1px solid #fff}.quickreply-wrapper{margin-top:5px}"]
    }),
    __metadata("design:paramtypes", [DomSanitizer])
], MessageComponent);

let UserInputComponent = 
/**
 * UserInputComponent class. This class wraps user message input box and Send SVG Component.
 * Inputs: options- Set styling options for UserInputComponent
 * @export
 * @class UserInputComponent
 * @implements {OnInit}
 */
class UserInputComponent {
    constructor() {
        /**
         * Event emitter for send button click
         * @memberof UserInputComponent
         */
        this.onInput = new EventEmitter();
    }
    ngOnInit() {
    }
    /**
     * Triggered when send button is clicked
     * Triggers OnInput event with input text as data if input text is not null
     * Clears the input text on event emit
     * @memberof UserInputComponent
     */
    send() {
        if (this.inputText) {
            this.onInput.emit(this.inputText);
            this.inputText = "";
        }
    }
};
__decorate([
    Output('onInput'),
    __metadata("design:type", Object)
], UserInputComponent.prototype, "onInput", void 0);
__decorate([
    Input('options'),
    __metadata("design:type", Object)
], UserInputComponent.prototype, "options", void 0);
UserInputComponent = __decorate([
    Component({
        selector: 'user-input',
        template: "<div class=\"input-wrapper\" [style.backgroundColor]=\"options?.backgroundColor\">\n    <input type=\"text\" [(ngModel)]=\"inputText\" placeholder=\"Say Something...\" \n    [attr.maxlength]=\"options?.maxLength\"\n    [style.borderColor]=\"options?.borderColor\"\n    [style.color]=\"options?.textColor\" (keyup.enter)=\"send()\"/>\n    <button class=\"btn-input-send\"  (click)=\"send()\">\n        <send-svg  [color]=\"options?.sendIconColor\"></send-svg>\n    </button>\n</div>",
        styles: [".input-wrapper{display:-webkit-box;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;flex-direction:row;padding-left:5px;height:35px;background-color:transparent}input{box-sizing:border-box;-webkit-box-flex:1;flex:1;border:1px solid #2196f3;background:0 0;color:#000;padding-left:5px;padding-right:5px;margin:2px;border-radius:15px}.btn-input-send{-webkit-box-flex:0;flex:0}button{background:0 0;border:none}::-webkit-input-placeholder{color:gray}:-ms-input-placeholder{color:gray}::-moz-placeholder{color:gray;opacity:.5}::-ms-input-placeholder{color:gray;opacity:.5}::placeholder{color:gray;opacity:.5}button:focus,input:focus,select:focus,textarea:focus{outline:0}send-svg{display:block;height:20px;width:20px;padding:5px 0;-webkit-transition:ease-in;transition:ease-in}send-svg:hover{height:18px;width:18px}"]
    })
    /**
     * UserInputComponent class. This class wraps user message input box and Send SVG Component.
     * Inputs: options- Set styling options for UserInputComponent
     * @export
     * @class UserInputComponent
     * @implements {OnInit}
     */
    ,
    __metadata("design:paramtypes", [])
], UserInputComponent);

let ChatHeaderComponent = class ChatHeaderComponent {
    /**
     *Creates an instance of ChatHeaderComponent.
     * @memberof ChatHeaderComponent
     */
    constructor() {
        this.title = 'Chat View';
    }
    ngOnInit() {
    }
};
__decorate([
    Input('title'),
    __metadata("design:type", Object)
], ChatHeaderComponent.prototype, "title", void 0);
__decorate([
    Input('headerOptions'),
    __metadata("design:type", Object)
], ChatHeaderComponent.prototype, "headerOptions", void 0);
ChatHeaderComponent = __decorate([
    Component({
        selector: 'chat-header',
        template: "<div class=\"chat-header-wrapper\" [style.backgroundColor]=\"headerOptions?.backgroundColor\">\n    <div class=\"chat-header-title\" [style.color]=\"headerOptions?.textColor\">\n        {{title}}\n    </div>\n</div>",
        styles: [".chat-header-wrapper{background-color:#2196f3;height:40px;display:-webkit-box;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;flex-direction:column;justify-content:space-around}.chat-header-title{text-align:center;vertical-align:middle;color:#fff}"]
    }),
    __metadata("design:paramtypes", [])
], ChatHeaderComponent);

let SendSVGImage = 
/**
 * SVG Image component for send icon image.
 * Color of this SVG component can be set dynamically using 'color' input property
 * @export
 * @class SendSVGImage
 * @implements {OnInit}
 */
class SendSVGImage {
    constructor() {
        this.color = "";
    }
    ngOnInit() {
    }
};
__decorate([
    Input('color'),
    __metadata("design:type", Object)
], SendSVGImage.prototype, "color", void 0);
SendSVGImage = __decorate([
    Component({
        selector: 'send-svg',
        template: "<svg enable-background=\"new 0 0 24 24\" height=\"100%\" viewBox=\"0 0 24 24\" width=\"100%\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"m8.75 17.612v4.638c0 .324.208.611.516.713.077.025.156.037.234.037.234 0 .46-.11.604-.306l2.713-3.692z\" [attr.fill]=\"color\"/><path d=\"m23.685.139c-.23-.163-.532-.185-.782-.054l-22.5 11.75c-.266.139-.423.423-.401.722.023.3.222.556.505.653l6.255 2.138 13.321-11.39-10.308 12.419 10.483 3.583c.078.026.16.04.242.04.136 0 .271-.037.39-.109.19-.116.319-.311.352-.53l2.75-18.5c.041-.28-.077-.558-.307-.722z\" [attr.fill]=\"color\"/></svg>",
        styles: [""]
    })
    /**
     * SVG Image component for send icon image.
     * Color of this SVG component can be set dynamically using 'color' input property
     * @export
     * @class SendSVGImage
     * @implements {OnInit}
     */
    ,
    __metadata("design:paramtypes", [])
], SendSVGImage);

let ChatViewModule = class ChatViewModule {
};
ChatViewModule = __decorate([
    NgModule({
        imports: [
            CommonModule,
            FormsModule
        ],
        declarations: [ChatViewComponent,
            ReceivedMessageComponent,
            SentMessageComponent,
            MessageComponent,
            UserInputComponent,
            ChatHeaderComponent,
            SendSVGImage
        ],
        exports: [ChatViewComponent]
    })
], ChatViewModule);

/*
 * Public API Surface of chat-view
 */

/**
 * Generated bundle index. Do not edit.
 */

export { ChatViewComponent, ChatViewModule, ChatViewService, MESSAGE_CATEGORY, MESSAGE_TYPES, Message, ReceivedMessage, SentMessage, messagePushAnimation as ɵa, ReceivedMessageComponent as ɵb, SentMessageComponent as ɵc, MessageComponent as ɵd, UserInputComponent as ɵe, ChatHeaderComponent as ɵf, SendSVGImage as ɵg };
