// chat-view/esm5/lib/components/message/message.component.js
import * as tslib_1 from "tslib";
import { Component, EventEmitter, HostBinding, HostListener, Input, Output } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { Message, MESSAGE_CATEGORY, ReceivedMessage } from '../message-data/message-data';
var MessageComponent = /** @class */ (function () {
    /**
     *Creates an instance of MessageComponent.
     * @param {DomSanitizer} sanitizer
     * @memberof MessageComponent
     */
    function MessageComponent(sanitizer) {
        this.sanitizer = sanitizer;
        /**
         * Output property which will emit an event on quick reply option is clicked.
         * @memberof MessageComponent
         */
        this.onOptionClicked = new EventEmitter();
        this.isRecivedMessage = false;
        this.quickReplies = [];
    }
    /**
     * Set 'isRecivedMessage' flag based on message category.
     * Assign values to text, image and quick replies variables.
     * Currently image and quick replies are only used for received message.
     * @memberof MessageComponent
     */
    MessageComponent.prototype.ngOnInit = function () {
        if (!this.message) {
            throw new Error('Data Required');
        }
        if (this.message.getCategory() == MESSAGE_CATEGORY.RECEIVED) {
            this.isRecivedMessage = true;
        }
        this.text = this.message.getText();
        this.image = this.message.getImageUrl();
        if (this.message instanceof ReceivedMessage) {
            this.quickReplies = this.message.getQuickReplies();
        }
    };
    /**
     * Emit 'onOptionClicked' event with 'option text' payload
     * @param {string} option
     * @memberof MessageComponent
     */
    MessageComponent.prototype.optionClicked = function (option) {
        this.onOptionClicked.emit(option);
    };
    /**
     * Set mouseover HostBinding for message component
     * This sets BoxShadow of component on 'hover'
     * @memberof MessageComponent
     */
    MessageComponent.prototype.onMouseOver = function () {
        if (this.isRecivedMessage) {
            this.boxShadow = this.sanitizer.bypassSecurityTrustStyle('1px 2px 8px rgba(33,33,33,.4)');
        }
        else {
            this.boxShadow = this.sanitizer.bypassSecurityTrustStyle('-2px 2px 8px rgba(33,33,33,.4)');
        }
    };
    /**
    * Set mouseleave HostBinding for message component
    * This removes BoxShadow of component.
    * @memberof MessageComponent
    */
    MessageComponent.prototype.onMouseLeave = function () {
        this.boxShadow = 'none';
    };
    MessageComponent.ctorParameters = function () { return [
        { type: DomSanitizer }
    ]; };
    tslib_1.__decorate([
        Input('data'),
        tslib_1.__metadata("design:type", Message)
    ], MessageComponent.prototype, "message", void 0);
    tslib_1.__decorate([
        Input('receivedMessageOptions'),
        tslib_1.__metadata("design:type", Object)
    ], MessageComponent.prototype, "receivedMessageOptions", void 0);
    tslib_1.__decorate([
        Input('sentMessageOptions'),
        tslib_1.__metadata("design:type", Object)
    ], MessageComponent.prototype, "sentMessageOptions", void 0);
    tslib_1.__decorate([
        Input('quickRepliesOptions'),
        tslib_1.__metadata("design:type", Object)
    ], MessageComponent.prototype, "quickRepliesOptions", void 0);
    tslib_1.__decorate([
        Output('onOptionClicked'),
        tslib_1.__metadata("design:type", Object)
    ], MessageComponent.prototype, "onOptionClicked", void 0);
    tslib_1.__decorate([
        HostBinding('style.boxShadow'),
        tslib_1.__metadata("design:type", Object)
    ], MessageComponent.prototype, "boxShadow", void 0);
    tslib_1.__decorate([
        HostListener('mouseover'),
        tslib_1.__metadata("design:type", Function),
        tslib_1.__metadata("design:paramtypes", []),
        tslib_1.__metadata("design:returntype", void 0)
    ], MessageComponent.prototype, "onMouseOver", null);
    tslib_1.__decorate([
        HostListener('mouseleave'),
        tslib_1.__metadata("design:type", Function),
        tslib_1.__metadata("design:paramtypes", []),
        tslib_1.__metadata("design:returntype", void 0)
    ], MessageComponent.prototype, "onMouseLeave", null);
    MessageComponent = tslib_1.__decorate([
        Component({
            selector: 'message',
            template: "<div *ngIf=\"message\">\n    <!--RECEIVED ITEM-->\n    <div class=\"message-wrapper\" *ngIf=\"isRecivedMessage\">\n\n        <div class=\"recv-text\" *ngIf=\"text\" [style.color]=\"receivedMessageOptions?.textColor\">{{text}}</div>\n\n        <div *ngIf=\"image\">\n            <img [src]=\"image\" alt=\"\">\n        </div>\n        <div class=\"quickreply-wrapper\" *ngIf=\"quickReplies && quickReplies.length>0\">\n            <div class=\"recv-quickreply\" *ngFor=\"let reply of quickReplies\"\n                [style.backgroundColor]=\"quickRepliesOptions?.backgroundColor\"\n                [style.borderColor]=\"quickRepliesOptions?.borderColor\">\n                <div (click)=\"optionClicked(reply)\" \n                [style.color]=\"quickRepliesOptions?.textColor\">{{reply}}</div>\n            </div>\n        </div>\n    </div>\n\n    <!--SENT ITEM-->\n    <div class=\"message-wrapper\" *ngIf=\"!isRecivedMessage\">\n        <div class=\"sent-text\" *ngIf=\"text\" [style.color]=\"sentMessageOptions?.textColor\">{{text}}</div>\n\n        <div *ngIf=\"image\">\n            <img [src]=\"image\" alt=\"\">\n        </div>\n    </div>\n</div>",
            styles: ["img{width:100%;display:block;margin:auto;max-width:95%;padding:5px;border-radius:10px}div{font-size:.9rem}.recv-text,.sent-text{word-wrap:break-word;padding:3px}.recv-quickreply{display:inline-block;border:1px solid rgba(255,255,255,.521);border-radius:10px;padding:3px 5px;margin-left:3px;cursor:pointer}.recv-quickreply:hover{box-shadow:0 0 11px rgba(33,33,33,.2);border:1px solid #fff}.quickreply-wrapper{margin-top:5px}"]
        }),
        tslib_1.__metadata("design:paramtypes", [DomSanitizer])
    ], MessageComponent);
    return MessageComponent;
}());
export { MessageComponent };
