// chat-view/esm5/lib/components/user-input/user-input.component.js
import * as tslib_1 from "tslib";
import { Component, EventEmitter, Input, Output } from '@angular/core';
var UserInputComponent = /** @class */ (function () {
    function UserInputComponent() {
        /**
         * Event emitter for send button click
         * @memberof UserInputComponent
         */
        this.onInput = new EventEmitter();
    }
    UserInputComponent.prototype.ngOnInit = function () {
    };
    /**
     * Triggered when send button is clicked
     * Triggers OnInput event with input text as data if input text is not null
     * Clears the input text on event emit
     * @memberof UserInputComponent
     */
    UserInputComponent.prototype.send = function () {
        if (this.inputText) {
            this.onInput.emit(this.inputText);
            this.inputText = "";
        }
    };
    tslib_1.__decorate([
        Output('onInput'),
        tslib_1.__metadata("design:type", Object)
    ], UserInputComponent.prototype, "onInput", void 0);
    tslib_1.__decorate([
        Input('options'),
        tslib_1.__metadata("design:type", Object)
    ], UserInputComponent.prototype, "options", void 0);
    UserInputComponent = tslib_1.__decorate([
        Component({
            selector: 'user-input',
            template: "<div class=\"input-wrapper\" [style.backgroundColor]=\"options?.backgroundColor\">\n    <input type=\"text\" [(ngModel)]=\"inputText\" placeholder=\"Say Something...\" \n    [attr.maxlength]=\"options?.maxLength\"\n    [style.borderColor]=\"options?.borderColor\"\n    [style.color]=\"options?.textColor\" (keyup.enter)=\"send()\"/>\n    <button class=\"btn-input-send\"  (click)=\"send()\">\n        <send-svg  [color]=\"options?.sendIconColor\"></send-svg>\n    </button>\n</div>",
            styles: [".input-wrapper{display:-webkit-box;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;flex-direction:row;padding-left:5px;height:35px;background-color:transparent}input{box-sizing:border-box;-webkit-box-flex:1;flex:1;border:1px solid #2196f3;background:0 0;color:#000;padding-left:5px;padding-right:5px;margin:2px;border-radius:15px}.btn-input-send{-webkit-box-flex:0;flex:0}button{background:0 0;border:none}::-webkit-input-placeholder{color:gray}:-ms-input-placeholder{color:gray}::-moz-placeholder{color:gray;opacity:.5}::-ms-input-placeholder{color:gray;opacity:.5}::placeholder{color:gray;opacity:.5}button:focus,input:focus,select:focus,textarea:focus{outline:0}send-svg{display:block;height:20px;width:20px;padding:5px 0;-webkit-transition:ease-in;transition:ease-in}send-svg:hover{height:18px;width:18px}"]
        })
        /**
         * UserInputComponent class. This class wraps user message input box and Send SVG Component.
         * Inputs: options- Set styling options for UserInputComponent
         * @export
         * @class UserInputComponent
         * @implements {OnInit}
         */
        ,
        tslib_1.__metadata("design:paramtypes", [])
    ], UserInputComponent);
    return UserInputComponent;
}());
export { UserInputComponent };
